src/ext/serializer.c

```diff
@@ -173,32 +175,125 @@ int ddtrace_serialize_simple_array(zval *trace, zval *retval TSRMLS_DC) {
 }
 
 #if PHP_VERSION_ID < 70000
-#define ADD_ELEMENT_IF_PROP_TYPE(name, type)                                                                    \
-    do {                                                                                                        \
-        zval *prop =                                                                                            \
-            zend_read_property(ddtrace_ce_span_data, span->span_data, (name), sizeof((name)) - 1, 1 TSRMLS_CC); \
-        if (Z_TYPE_P(prop) == (type)) {                                                                         \
-            zval *value;                                                                                        \
-            ALLOC_ZVAL(value);                                                                                  \
-            INIT_PZVAL_COPY(value, prop);                                                                       \
-            zval_copy_ctor(value);                                                                              \
-            add_assoc_zval(el, (name), value);                                                                  \
-        }                                                                                                       \
-    } while (0);
+static zval *_read_span_property(zval *span_data, const char *name, size_t name_len TSRMLS_DC) {
+    return zend_read_property(ddtrace_ce_span_data, span_data, name, name_len, 1 TSRMLS_CC);
+}
+
+static void _add_assoc_zval(zval *el, const char *name, zval *prop) {
+    zval *value;
+    ALLOC_ZVAL(value);
+    INIT_PZVAL_COPY(value, prop);
+    zval_copy_ctor(value);
+    add_assoc_zval(el, name, value);
+}
+
+void _serialize_exception(zval *el, zval *meta, ddtrace_span_t *span TSRMLS_DC) {
+    zval *exception, *msg, *stack;
+    zend_uint class_name_len;
+    const char *class_name;
+    int dup;
```

Comment:
What value this is set to ?

Reply:
It's set on line 199; I forward declare my local variables out of habit in php-src since that was a requirement for a long time (not even sure if it's still a requirement -- IIRC Windows toolchain was last holdout).

Reply:
:| Missed it - thanks. 
In our case we don't need to forward declare - usually I'm fine with either. 
Here the dup escaped my brain parser - so it might indicate, either broken parser, or visibility problem.

Not to bikeshed too much - I think `needs_duplication` might be more descriptive name. Especially when its a result from function named `zend_get_object_classname`. 

Both work though, so lets treat  this as optional suggestion.

Reply:
I moved definition of `dup` to its initialization, which should help.